src/Cli/AzureCliLogin.ts

```diff
@@ -86,22 +88,67 @@ export class AzureCliLogin {
-            await this.executeAzCliCommand(`cloud register -n "${this.loginConfig.environment}" --endpoint-resource-manager "${this.loginConfig.resourceManagerEndpointUrl}" --suffix-keyvault-dns "${suffixKeyvault}" --suffix-storage-endpoint "${suffixStorage}" --profile "${profileVersion}"`, false);
+            await this.executeAzCliCommand(["cloud", "register", "-n", this.loginConfig.environment, "--endpoint-resource-manager", `"${this.loginConfig.resourceManagerEndpointUrl}"`, "--suffix-keyvault-dns", `"${suffixKeyvault}"`, "--suffix-storage-endpoint", `"${suffixStorage}"`, "--profile", `"${profileVersion}"`], false);
         }
         catch (error) {
-            core.error(`Error while trying to register cloud "${this.loginConfig.environment}": "${error}"`);
+            core.error(`Error while trying to register cloud "${this.loginConfig.environment}"`);
+            throw error;
+        }
+
+        core.info(`Done registering cloud: "${this.loginConfig.environment}"`)
+    }
+
+    async loginWithSecret(args: string[]) {
+        core.info("Note: Azure/login action also supports OIDC login mechanism. Refer https://github.com/azure/login#configure-a-service-principal-with-a-federated-credential-to-use-oidc-based-authentication for more details.")
+        args.push(`--password=${this.loginConfig.servicePrincipalKey}`);
+        await this.callCliLogin(args, 'service principal with secret');
+    }
+
+    async loginWithOIDC(args: string[]) {
+        await this.loginConfig.getFederatedToken();
+        args.push("--federated-token", this.loginConfig.federatedToken);
+        await this.callCliLogin(args, 'OIDC');
+    }
+
+    async loginWithUserAssignedIdentity(args: string[]) {
+        args.push("--username", this.loginConfig.servicePrincipalId);
+        await this.callCliLogin(args, 'user-assigned managed identity');
+    }
+
+    async loginWithSystemAssignedIdentity(args: string[]) {
+        await this.callCliLogin(args, 'system-assigned managed identity');
+    }
+
+    async callCliLogin(args: string[], methodName: string) {
+        core.info(`Attempting Azure CLI login by using ${methodName}...`);
+        args.unshift("login");
+        if (this.loginConfig.allowNoSubscriptionsLogin) {
+            args.push("--allow-no-subscriptions");
         }
+        await this.executeAzCliCommand(args, true, this.loginOptions);
+        await this.setSubscription();
+        core.info(`Azure CLI login succeeds by using ${methodName}.`);
+    }
 
-        console.log(`Done registering cloud: "${this.loginConfig.environment}"`)
+    async setSubscription() {
+        if (this.loginConfig.allowNoSubscriptionsLogin) {
+            return;
+        }
+        if (!this.loginConfig.subscriptionId) {
+            core.warning('No subscription-id is given. Skip setting subscription... If there are mutiple subscriptions under the tenant, please input subscription-id to specify which subscription to use.');

```

Comment:
> No subscription-id is given. Skip setting subscription...

This statement is not accurate. Azure CLI chooses the first active subscription as the default/subscription, so a implicit setting subscription is performed.